### PR TITLE
ARROW-2170: [Python] construct_metadata fails on reading files where no index was preserved

### PR DESCRIPTION
**Pull request by xhochy**
cc @cpcloud 

The result was that we only persisted empty Pandas metadata for these files.

### CODE REVIEW

**Comment by cpcloud** [comment]
@xhochy If you rebase on top of master you should pick up ARROW-2169 which will allow the appveyor build to pass.

File: python/pyarrow/pandas_compat.py [hunk 1]
```diff
@@ -197,8 +197,11 @@ def construct_metadata(df, column_names, index_levels, index_column_names,
     -------
     dict
     """
-    df_types = types[:-len(index_levels)]
-    index_types = types[-len(index_levels):]
+    # Use ntypes instead of Python shorthand notation [:-len(x)] as [:-0]
+    # behaves differently to what we want.
```
Comment (cpcloud) [inline python/pyarrow/pandas_compat.py]: Nice catch.

File: python/pyarrow/tests/test_parquet.py [hunk 2]
```diff
@@ -215,6 +215,9 @@ def test_pandas_parquet_2_0_rountrip_read_pandas_no_index_written(tmpdir):
     arrow_table = pa.Table.from_pandas(df, preserve_index=False)
     js = json.loads(arrow_table.schema.metadata[b'pandas'].decode('utf8'))
     assert not js['index_columns']
+    # ARROW-2170
+    # While index_columns should be empty, columns needs to be filled still.
+    assert len(js['columns']) > 0
```
Comment (cpcloud) [inline python/pyarrow/tests/test_parquet.py]: You can just do `assert js['columns']` here

**Review by cpcloud — APPROVED** [review]
LGTM

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**